python/spark_sklearn/grid_search.py

```diff
@@ -10,8 +10,123 @@
+        par_param_grid = self.sc.parallelize(indexed_param_grid, len(indexed_param_grid))
+        X_bc = self.sc.broadcast(X)
+        y_bc = self.sc.broadcast(y)
+
+        scorer = self.scorer_
+        verbose = self.verbose
+        fit_params = self.fit_params
+        error_score = self.error_score
+        fas = _fit_and_score
+
+        def fun(tup):
+            (index, (parameters, train, test)) = tup
+            local_estimator = clone(base_estimator)
+            local_X = X_bc.value
+            local_y = y_bc.value
+            res = fas(local_estimator, local_X, local_y, scorer, train, test, verbose,
+                                  parameters, fit_params,
+                                  return_parameters=True, error_score=error_score)
+            return (index, res)
+        indexed_out0 = dict(par_param_grid.map(fun).collect())
+        out = [indexed_out0[idx] for idx in range(len(param_grid))]
+
+        X_bc.unpersist()
+        y_bc.unpersist()
+
+        # Out is a list of triplet: score, estimator, n_test_samples
+        n_fits = len(out)
+        n_folds = len(cv)
+
+        scores = list()
+        grid_scores = list()
+        for grid_start in range(0, n_fits, n_folds):
+            n_test_samples = 0
+            score = 0
+            all_scores = []
+            for this_score, this_n_test_samples, _, parameters in \
+                    out[grid_start:grid_start + n_folds]:
+                all_scores.append(this_score)
+                if self.iid:
+                    this_score *= this_n_test_samples
+                    n_test_samples += this_n_test_samples
+                score += this_score
+            if self.iid:
+                score /= float(n_test_samples)
+            else:
+                score /= float(n_folds)
+            scores.append((score, parameters))
+            # TODO: shall we also store the test_fold_sizes?
+            grid_scores.append(_CVScoreTuple(
+                parameters,
+                score,
+                np.array(all_scores)))
+        # Store the computed scores
+        self.grid_scores_ = grid_scores
+
+        # Find the best parameters by comparing on the mean validation score:
+        # note that `sorted` is deterministic in the way it breaks ties
+        best = sorted(grid_scores, key=lambda x: x.mean_validation_score,
+                      reverse=True)[0]
+        self.best_params_ = best.parameters
+        self.best_score_ = best.mean_validation_score
+
+        if self.refit:
+            # fit the best estimator using the entire dataset
+            # clone first to work around broken estimators
+            best_estimator = clone(base_estimator).set_params(
+                **best.parameters)
+            if y is not None:
+                best_estimator.fit(X, y, **self.fit_params)
+            else:
+                best_estimator.fit(X, **self.fit_params)
+            self.best_estimator_ = best_estimator
```

Comment:
Where is `cv_results_` exists in current master version ?
It seems that you change a lot here.